runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -7067,17 +7060,10 @@ int32_t setUpHooks(J9JavaVM * javaVM, J9JITConfig * jitConfig, TR_FrontEnd * vm)
 
       IDATA unableToRegisterHooks = 0;
 
-      if (TR::Options::getCmdLineOptions()->realTimeGC())
-         {
-         if (!TR::Options::getCmdLineOptions()->getOption(TR_NoClassGC))
-            unableToRegisterHooks = (*gcOmrHooks)->J9HookRegisterWithCallSite(gcOmrHooks, J9HOOK_MM_OMR_GC_CYCLE_END, jitHookReleaseCodeGCCycleEnd, OMR_GET_CALLSITE(), NULL);
-         }
-      else
-         {
-         unableToRegisterHooks =
-             ((*gcOmrHooks)->J9HookRegisterWithCallSite(gcOmrHooks, J9HOOK_MM_OMR_GLOBAL_GC_END, jitHookReleaseCodeGlobalGCEnd, OMR_GET_CALLSITE(), NULL) ||
-             (*gcOmrHooks)->J9HookRegisterWithCallSite(gcOmrHooks, J9HOOK_MM_OMR_LOCAL_GC_END, jitHookReleaseCodeLocalGCEnd, OMR_GET_CALLSITE(), NULL));
-         }
+
+      if (!TR::Options::getCmdLineOptions()->realTimeGC() || !TR::Options::getCmdLineOptions()->getOption(TR_NoClassGC)) {
+         unableToRegisterHooks = (*gcOmrHooks)->J9HookRegisterWithCallSite(gcOmrHooks, J9HOOK_MM_OMR_GC_CYCLE_END, jitHookReleaseCodeGCCycleEnd, OMR_GET_CALLSITE(), NULL);
```

Comment:
Here is my understanding of the state of the world going forward.  Is this correct?  I might have further questions depending on the clarification of my understanding.

* for <JDK 19, all thread stack walks for code cache reclamation will now be triggered by the GC_CYCLE_END event
* for >=JDK19, all platform thread stack walks for code cache reclamation will now be triggered by the GC_CYCLE_END event.  Virtual thread stack walks will now be triggered by a new WALKCONTINUATION event, and this is the only way virtual thread stacks are walked.

Reply:
`  for <JDK 19, all thread stack walks for code cache reclamation will now be triggered by the GC_CYCLE_END event
` 
yes
`  for >=JDK19, all platform thread stack walks for code cache reclamation will now be triggered by the GC_CYCLE_END event. Virtual thread stack walks will now be triggered by a new WALKCONTINUATION event, and this is the only way virtual thread stacks are walked.`
yes, since Virtual thread stack walks is the most expensive, so move it to GC clearable phase(via new method iterateAllContinuationObjects(), parallel running by all of GC threads), the gc clearable phase is guaranteed before GC_CYCLE_END.